docs/src/main/sphinx/connector/jdbc-common-configurations.fragment

```diff
@@ -51,3 +51,7 @@ connector:
       Using a large timeout can potentially result in more detailed dynamic filters.
       However, it can also increase latency for some queries.
     - ``20s``
+  * - ``query.reuse-connection``
```

Comment:
This is a kill switch, added only in case of bugs. I would prefer not document this and possibly remove in future once we see there are no issues related to this new feature.

Reply:
It was added to the release notes...

Reply:
I would say that if users run into issues, ideally they should be able to see the potentially figure out how to work around it by looking at connector docs, not by looking at release notes.

If/when we decide to remove the property, it should be easy to also remove the docs at that juncture, yeah?

Reply:
I don't think we document all the kill switches.

Reply:
So how are users supposed to know about it, if we don't document or expose it? I think we should document them. And if we want to remove them at some stage .. we can. 

Or are you saying you want to prevent users from knowing about it and/or using them. If so .. then why did we add it in the first place. Purely for developers?

Reply:
It is here, so in case we hear a feedback that it does not work then we may want to suggest to use it. I plan to remove the kill switch after some time if there is no feedback. 

I we place this in documentation then we may not hear the feedback at all and we could just have bugs that people bypass while I would like to fix these bugs.

Reply:
the problem with documenting kill switches is that users enable those to avoid change (I know I did) and then you can never know if people actually use the new code and then one day you remove kill switch and people start filing bugs that thing was broken.

Reply:
So here is what I would do. Always do one or the other:

* implement a kill switch and document it
* don't implement a kill switch

If you need to later remove a kill switch .. you could leave it behind as a no-op with a warning for a while and then remove it.